File: Client/app/types/job.ts
export interface JobImageCreateRequest {
  filePath: string;
  fileName?: string;
  fileType?: string;
  fileSize?: number;
  caption?: string;
  sortOrder?: number;
  isPrimary?: boolean;
  isActive?: number;
}

export interface JobCreateRequest {
  title: string;
  description?: string;
  employmentType?: string;
  salaryFrom?: number;
  salaryTo?: number;
  positionsNeeded: number;
  positionsFilled: number;
  deadlineDate?: string; // ISO date string
  minAge?: number;
  maxAge?: number;
  requiredExperienceYears?: number;
  requiredDegree?: string;
  genderPreference?: "Nam" | "Nữ" | "Không yêu cầu";
  skillsRequired?: string;
  categoryId: string; // This should be a valid GUID string
  companyId?: string;
  images?: JobImageCreateRequest[];
}

export interface JobDto {
  jobId: string;
  jobGuid: string;
  title: string;
  description?: string;
  employmentType?: string;
  salaryFrom?: number;
  salaryTo?: number;
  createdAt: string;
  hiringStatus?: string;
  positionsNeeded?: number;
  positionsFilled?: number;
  deadlineDate?: string;
  minAge?: number;
  maxAge?: number;
  requiredExperienceYears?: number;
  requiredDegree?: string;
  genderPreference?: string;
  skillsRequired?: string;
  categoryId?: string;
  imageUrl?: string;
  isActive?: number;
  company?: {
    companyId: string;
    name: string;
    logoURL?: string;
  };
  images?: {
    imageId: string;
    filePath: string;
    isPrimary: boolean;
  }[];
}

export interface CategoryDto {
  categoryId: string;
  name: string;
  description?: string;
}

File: Server/Server/Models/Jobs/Job.cs
using System;
using Server.Models.Profiles;

namespace Server.Models.Jobs
{
    public class Job
    {
        public Guid JobId { get; set; }
        public Guid JobGuid { get; set; }
        public Guid PostedByUserId { get; set; }
        public Guid? EmployerProfileId { get; set; }
        public Guid? CompanyId { get; set; }
        public string Title { get; set; } = string.Empty;
        public string? Description { get; set; }
        public string? EmploymentType { get; set; }
        public decimal? SalaryFrom { get; set; }
        public decimal? SalaryTo { get; set; }
        public int IsActive { get; set; }
        public DateTime CreatedAt { get; set; }
        public string? HiringStatus { get; set; }
        public int? PositionsNeeded { get; set; }
        public int? PositionsFilled { get; set; }
        public DateTime? DeadlineDate { get; set; }
        public int? MinAge { get; set; }
        public int? MaxAge { get; set; }
        public int? RequiredExperienceYears { get; set; }
        public string? RequiredDegree { get; set; }
        public string? GenderPreference { get; set; }
        public string? SkillsRequired { get; set; }
        public Guid? CategoryId { get; set; }

        // Navigation property
        public Company? Company { get; set; }
    }
}
